test/e2e/e2e_suite_test.go

```diff
@@ -84,6 +86,16 @@ var (
 	bootstrapClusterProxy framework.ClusterProxy
 
 	namespaces map[*corev1.Namespace]context.CancelFunc
+
+	// e2eIPAMKubeconfig is a kubeconfig to a cluster which provides IP address management via an in-cluster
+	// IPAM provider to claim IPs for the control plane IPs of created clusters.
+	e2eIPAMKubeconfig string
+
+	// testSpecificClusterctlConfigPath gets written before running a test and is a copy of the clsuterctlConfigPath with
+	// the addition of variables for control plane IP addresses.
+	testSpecificClusterctlConfigPath string
+	// testSpecificIPAMCleanupFunc is used to cleanup IP addresses after successfully running a test.
+	testSpecificIPAMCleanupFunc func()
```

Comment:
I would prefer if we keep these as variables with local scope instead of having package global variables.

I guess it works because we never run more than one test sequentially in one process, but it feels a bit dirty to continuously update package global variables when going through the tests.